js/completionCalculations.js

```diff
@@ -5,144 +5,152 @@ define([
+
+                if (!contentObjectModel.get('_pageLevelProgress')._excludeAssessments) {
+                    subProgressCompleted = contentObjectModel.get("_subProgressComplete") || 0;
+                    subProgressTotal = contentObjectModel.get("_subProgressTotal") || 0;
+                }
+
+                var pageCompletion = {
+                    "subProgressCompleted": subProgressCompleted,
+                    "subProgressTotal": subProgressTotal,
+                    "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
+                    "nonAssessmentTotal": nonAssessmentComponentsTotal,
+                    "assessmentCompleted": assessmentComponentsCompleted,
+                    "assessmentTotal": assessmentComponentsTotal
+                };
+
+                var showPageCompletionCourse = Adapt.course.get("_pageLevelProgress") && Adapt.course.get("_pageLevelProgress")._showPageCompletion !== false;
+                var showPageCompletionPage = contentObjectModel.get("_pageLevelProgress") && contentObjectModel.get("_pageLevelProgress")._showPageCompletion !== false;
+
+                if (showPageCompletionCourse && showPageCompletionPage) {
+                    // optionally add one point extra for page completion to eliminate incomplete pages and full progress bars
+                    // if _showPageCompletion is true then the progress bar should also consider it so add 1 to nonAssessmentTotal
+                    pageCompletion.nonAssessmentCompleted += isComplete;
+                    pageCompletion.nonAssessmentTotal += 1;
+                }
+
+                return pageCompletion;
+            case 'menu':
+                // If it's a sub-menu
+                children = contentObjectModel.get('_children').models;
+                children.forEach(function(contentObject) {
+                    var completionObject = calculateCompletion(contentObject);
+                    subProgressCompleted += contentObjectModel.subProgressCompleted || 0;
+                    subProgressTotal += contentObjectModel.subProgressTotal || 0;
+                    nonAssessmentComponentsTotal += completionObject.nonAssessmentTotal;
+                    nonAssessmentComponentsCompleted += completionObject.nonAssessmentCompleted;
+                    assessmentComponentsTotal += completionObject.assessmentTotal;
+                    assessmentComponentsCompleted += completionObject.assessmentCompleted;
+                });
+
+                return {
+                    "subProgressCompleted": subProgressCompleted,
+                    "subProgressTotal" : subProgressTotal,
+                    "nonAssessmentCompleted": nonAssessmentComponentsCompleted,
+                    "nonAssessmentTotal": nonAssessmentComponentsTotal,
+                    "assessmentCompleted": assessmentComponentsCompleted,
+                    "assessmentTotal": assessmentComponentsTotal,
```

Comment:
Double quotes.

Reply:
I was also wondering if, since this data is always returned as an object, would it not be worth declaring it as an object from the outset?